### PR TITLE
Form Input revamp for bundle size reduction

### PR DESCRIPTION
**Pull request by kd4001**

### CODE REVIEW

File: src/components/generic/form/FormInput.re [hunk 1]
```diff
@@ -1,16 +1,40 @@
...
+  result,
+  warning,
+  className,
+  _inputProps,
+};
+
+module Date = {
+  include FormDateInput;
+};
+
```
Comment (Prateek13727) [inline src/components/generic/form/FormInput.re:34]: lets take type  formInput our of this module into a module of its own

File: src/components/generic/form/FormInput.re [hunk 2]
```diff
@@ -1,16 +1,40 @@
...
 type formInput =
   | Input(string)
```
Comment (Prateek13727) [inline src/components/generic/form/FormInput.re:36]: also, create a folder form-input inside form and move the FormInput and helper modules inside it

File: src/components/generic/form/FormInput.re [hunk 3]
```diff
@@ -1,16 +1,40 @@
...
-  | Date({
```
Comment (Prateek13727) [inline src/components/generic/form/FormInput.re:7]: remove getWarningOrError from this module

File: src/components/generic/form/FormInputWrapper.re [hunk 4]
```diff
@@ -0,0 +1,22 @@
+let getWarningOrError =
```
Comment (Prateek13727) [inline src/components/generic/form/FormInputWrapper.re:1]: make a module `FormInputHelper` and move `getWarningOrError` function there. this is more of a utility function, does not make sense in the wrapper component

Comment (Prateek13727) [reply src/components/generic/form/FormInputWrapper.re:1]: also move `get_InputProps` from `FormInput` to `FormInputHelper`

File: src/components/generic/form/FormDateInput.re [hunk 5]
```diff
@@ -0,0 +1,27 @@
+type date_props = {
```
Comment (Prateek13727) [inline src/components/generic/form/FormDateInput.re:1]: rename `type date-props` to `type t`

File: src/components/generic/form/FormInput.re [hunk 6]
```diff
@@ -1,16 +1,40 @@
 open Belt;
+open FormInputProps;
 type result('a, 'b) = Result.t('a, 'b);
 let error = err => Result.Error(err);
```
Comment (Prateek13727) [inline src/components/generic/form/FormInput.re:4]: is `let error = err => Result.Error(err)` this being used somewhere ?

File: src/components/generic/form/FormInput.re (hunk 1 above)
Comment (kd4001) [reply src/components/generic/form/FormInput.re:6]: Moving this out would lead to importing every single module into that module. 
Like FormInputDate will have to be imported into that

File: package.json [hunk 7]
```diff
@@ -30,18 +30,19 @@
...
     "@material-ui/pickers": "^3.2.10",
     "@sentry/browser": "^5.3.0",
+    "react-ga": "^3.3.0",
     "bs-fetch": "^0.6.2",
     "bs-moment": "^0.4.4",
-    "bs-webapi": "^0.17.1",
-    "decco": "^1.1.1",
     "moment": "^2.24.0",
-    "papaparse": "5.1.0",
+    "react-datepicker": "^2.9.6",
```
Comment (Prateek13727) [inline package.json:37]: which comp does this dependency `react-date picker ` maps to ?

File: bsconfig.json [hunk 8]
```diff
@@ -15,11 +15,17 @@
     "ppx-flags": [
       "decco/ppx"
     ],
-    "bs-dev-dependencies": [],
+    "bs-dev-dependencies": [
```
Comment (Prateek13727) [inline bsconfig.json:18]: we need to revert these changes also ?

File: package.json (hunk 7 above)
Comment (kd4001) [reply package.json:37]: It is added in main.scss. Css of this is being used. Don't know where

Comment (kd4001) [reply package.json:37]: Let me confirm

File: yarn.lock [hunk 9]
```diff
@@ -2,7 +2,7 @@
 # yarn lockfile v1
 
 
-"@babel/runtime@^7.3.1", "@babel/runtime@^7.4.4", "@babel/runtime@^7.5.5", "@babel/runtime@^7.6.0", "@babel/runtime@^7.8.3", "@babel/runtime@^7.8.7":
+"@babel/runtime@^7.1.2", "@babel/runtime@^7.3.1", "@babel/runtime@^7.4.4", "@babel/runtime@^7.5.5", "@babel/runtime@^7.6.0", "@babel/runtime@^7.8.3", "@babel/runtime@^7.8.7":
```
Comment (Prateek13727) [inline yarn.lock:5]: yarn lock ke changes bhi  revert kar dena

Comment (Prateek13727) [reply yarn.lock:5]: this has changes from removed dependencies
